### DISCUSSION TITLE
Is there a way to add a class to my custom element on validation error?

### DISCUSSION THREAD

**Discussion by deanCompRSA** [vueform/vueform#296] [Questions] [answered]
I basically want to add a red border when there is an error. Is there a way to add a class on error? Or to check for a validation error inside the element?

**Comment by adamberecz** [comment, marked as answer]
If it is a custom element I'd recommend checking out this part:
https://vueform.com/docs/creating-elements#classes

Eg. if you want to assign dynamic classes based on component state to `wrapper` you can use `$wrapper`:

```js
defaultClasses: {
  wrapper: 'border',
  wrapper_no_error: 'border-gray-500',
  wrapper_error: 'border-red-500',
  $wrapper: (classes, ({ error }) => ([
    classes.wrapper,
    error ? classes.wrapper_error : classes.wrapper_no_error,
  ]))
}
```

The first arg of the func is the classes object of the component, the second is the component instance where you can access any of its props.
